build/Microsoft.DotNet.Cli.Compile.targets

```diff
@@ -0,0 +1,182 @@
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="compile/Microsoft.DotNet.Cli.LzmaArchive.targets" />
+
+  <UsingTask TaskName="FixModeFlags" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="ChangeEntryPointLibraryName" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="RemoveAssetFromDepsPackages" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="CleanPublishOutput" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="CrossgenDirectory" AssemblyFile="$(CLIBuildDll)" />    
+
+  <Target Name="Compile" DependsOnTargets="Init;
+                                           SetupStage;
+                                           CompileStage;
+                                           BuildProjectsForNuGetPackages;
+                                           GenerateNuGetPackagesArchive">
+  </Target>
+
+  <Target Name="SetupStage">
+    <PropertyGroup>
+      <SrcDirectory>$(RepoRoot)/src</SrcDirectory>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <CompileStageInputs Include="$(SrcDirectory)/**/*.cs" Exclude="$(SrcDirectory)/**/obj/**/*.cs" />
+      <CompileStageInputs Include="$(SrcDirectory)/**/project.json;$(SrcDirectory)/**/project.lock.json" />
```

Comment:
This makes me super nervous. This is not an authoritative. The ideal would be to have .csproj's calculate their own incrementality. However, if we want a cheap solution for the time being, can we just `Include="$(SrcDirectory)/**/*.*"` with no excludes? E.g. if anything in src changes, assume you need to update the output. This seems much less prone to assumptions.